src/main/java/com/ably/kafka/connect/offset/OffsetRegistry.java

```diff
@@ -0,0 +1,12 @@
+package com.ably.kafka.connect.offset;
+
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.connect.sink.SinkRecord;
+
+import java.util.List;
+
+public interface OffsetRegistry {
```

Comment:
I would just update this interface definition to be the two functions you actually need and delete the original proposals. You can this pass this type around instead of the implementation class.